test/Microsoft.TemplateSearch.Common.UnitTests/TemplateSearchCacheReaderTests.cs

```diff
@@ -171,5 +171,30 @@ public void CanReadSearchMetadata_V2_E2E()
             Assert.Equal(2, ((ITemplateInfo)templateToTest).PostActions.Count);
             Assert.Equal(new[] { postAction1, postAction2 }, ((ITemplateInfo)templateToTest).PostActions);
         }
+
+        private async Task<string> AttemptGetSearchFileAsync(int count, int intervalInSeconds, NuGetMetadataSearchProvider searchProvider)
```

Comment:
`AttemptSearchAsync` and `AttemptGetSearchFileAsync` can be unified into single helper function taking a functor and nonretry exception type